party.py

```diff
@@ -0,0 +1,102 @@
+import roles
+import discord
+from datetime import datetime
+
+# 86400 seconds in 24 hours
+# 43200 seconds in 12 hours
+# 3600 seconds in an hour
+ACTIVE_DURATION_SECONDS = 86400
+DEFAULT_PARTY_SIZE = 4
+DEFAULT_JOIN_EMOJI = '👍'
+DEFAULT_CLOSE_EMOJI = '❌'
+
+class party:
+    def __init__(self, message, leader, name, size=None):
+        self.__close = False
+        self.__message = message
+        self.__leader = leader
+        self.__creationDateTime = datetime.now()
+        self.__partyList = [leader.name]
+        self.__waitlist = []
+
+        preset = self.__getPreset(name)
+
+        if (preset is not None):
+            self.name = preset.name
+            self.size = preset.size if size is None else size
+            self.imageURL = preset.imageURL
+            self.joinEmoji = preset.emoji
+        else:
+            self.name = name
+            self.size = DEFAULT_PARTY_SIZE if size is None else size
+            self.imageURL = None
+            self.joinEmoji = DEFAULT_JOIN_EMOJI
+
+        self.closeEmoji = DEFAULT_CLOSE_EMOJI
+
+    @staticmethod
+    def __getPreset(name):
+        for preset in roles.ROLES_LIST:
+            if name.lower() == preset.name.lower():
```

Comment:
casefold for safety? I dunno. https://stackoverflow.com/questions/45745661/python-lower-vs-casefold-in-string-matching-and-converting-to-lowercase

Reply:
casefold sounds good. i'm not sure if roles can have emoji's, but let's just opt for safer